extensions/http-curl/tests/TestServer.h

```diff
@@ -24,65 +24,87 @@
 #include "CivetServer.h"
 #include "civetweb.h"
 #include "HTTPUtils.h"
+#include "ServerAwareHandler.h"
 
-
-/* Server context handle */
-static std::string resp_str;
-
-void init_webserver() {
-  mg_init_library(0);
-}
-
-
-CivetServer * start_webserver(std::string &port, std::string &rooturi, CivetHandler *handler, struct mg_callbacks *callbacks, std::string &cert, std::string &ca_cert) {
-  const char *options[] = { "document_root", ".", "listening_ports", port.c_str(), "error_log_file",
-      "error.log", "ssl_certificate", ca_cert.c_str(), "ssl_protocol_version", "4", "ssl_cipher_list",
-      "ALL", "request_timeout_ms", "10000", "enable_auth_domain_check", "no", "ssl_verify_peer", "no", 0 };
-// ECDH+AESGCM+AES256:!aNULL:!MD5:!DSS
-  std::vector<std::string> cpp_options;
-  for (size_t i = 0; i < (sizeof(options) / sizeof(options[0]) - 1); i++) {
-    cpp_options.push_back(options[i]);
-  }
-
-  if (!mg_check_feature(2)) {
-      fprintf(stderr,
-              "Error: Embedded example built with SSL support, "
-              "but civetweb library build without.\n");
-      return 0;
+/**
+ * A wrapper around CivetServer which notifies Handlers before shutdown,
+ * so they wouldn't get stuck (if a handler returns after shutdown is
+ * initiated it might get stuck inside worker_thread_run > consume_socket)
+ */
+class TestServer{
+  struct CivetLibrary{
+    CivetLibrary() {
+      if (getCounter()++ == 0) {
+        mg_init_library(0);
+      }
+    }
+    ~CivetLibrary() {
+      if (--getCounter() == 0) {
```

Comment:
What's the motivation behind using a getter on our own private member?

Reply:
will make the `getCounter` static, we do not access a member though this getter, but a static library initialization counter

Reply:
done